tests/src/androidTest/java/com/squareup/duktape/DuktapeGetTest.java

```diff
@@ -388,4 +388,189 @@
     assertThat(summer.countTrues()).isEqualTo(0.0);
     assertThat(summer.countTrues(true, false, true, true)).isEqualTo(3.0);
   }
+
+  interface ObjectSorter {
+    Object[] sort(Object[] args);
+  }
+
+  private static final String SORTER_FUNCTOR = ""
+      + "var Sorter = {\n"
+      + "  sort: function(v) {"
+      + "    if (v) {\n"
+      + "      v.sort();\n"
+      + "    }\n"
+      + "    return v;\n"
+      + "  },\n"
+      + "  sortNullable: function(v) { return this.sort(v); }\n"
+      + "};";
+
+  @Ignore // TODO: rework array marshalling to not use the Duktape stack.
+  @Test public void longArrays() {
+    duktape.evaluate(SORTER_FUNCTOR);
+
+    ObjectSorter sorter = duktape.get("Sorter", ObjectSorter.class);
+
+    int length = 10000; // Current safe limit is ~249.
+    assertThat(sorter.sort(new Object[length])).hasLength(length);
+  }
+
+  @Test public void arraysOfObjects() {
+    duktape.evaluate(SORTER_FUNCTOR);
+
+    ObjectSorter sorter = duktape.get("Sorter", ObjectSorter.class);
+
+    assertThat(sorter.sort(null)).isNull();
+
+    Object[] original = new Object[]{2, 4, 3, 1};
+    Object[] sorted = sorter.sort(original);
+    assertThat(sorted).isEqualTo(new Object[]{1.0, 2.0, 3.0, 4.0});
```

Comment:
We can't tell the difference between an int and double in JS, so we treat everything as a double (this is related to why we chose not to support int as a return value to Java).